### ISSUE TITLE
255 cartographic mathematical data

### ISSUE THREAD

**Issue by CECSpecialistI** [uwlib-cams/MARC2RDA#121]
https://github.com/uwlib-cams/MARC2RDA/blob/main/Working%20Documents/2XX.csv

**Comment by dchen077** [comment]
Subfield d Statement of Zone is mapped to P10082 right ascension and declination and the transform note indicates "record the declination limits of each zone. Omit the statement of right ascension." Some of the examples given in OCLC and LOC are `$d(Right ascension 0 hr. to 24 hr./Declination +90° to -90°)` or `$d(RA 16 hr./Decl. +30° ;$eeq. 1973.50). `

I just want to confirm if the transformed result is intended to be something like `<P10082>Declination +90° to -90°</P10082>` or `0 hr. to 24 hr./Declination +90° to -90°` ?

Furthermore, if we should strip the parenthesis or punctuation, especially since $e is included in the parenthesis of $d? I am not sure if that is going to cause any issues after the transform.

**Comment by pennylenger** [comment]
Hi @dchen077, I now don't think "Omit the statement of right ascension" is correct. We should keep the right ascension data as well as the  punctuation.

**Comment by tynancuw** [comment]
Hey @cspayne, I have pushed my first pass at the code for 255 up to the git repo. Are you able to review this work? I ran a test and the append labels scenario. Thanks!

**Comment by cspayne** [comment]
Hi @tynancuw 
This looks great! I have a few comments: 
1. For subfield a, you are currently using `contains(., '[0-9]')` and `not(contains(., '[0-9]'))`. `contains()` checks whether a string value contains another string value, so you are currently checking whether the subfield contains the exact string "[0-9]". Instead, you want to use `matches()`, which checks whether a string matches a regular expression. This will instead check whether $a contains any digits between 0 and 9, which is what we want. 
2. The `replace()` function for subfield c looks good, however, you may want to add trailing whitespace in the regular expression just in case, something like `^\(|\)$|\).$|\.\s$`. Another option is to `normalize-space()` both before and after the `replace()` function: `normalize-space(marc:subfield[@code = 'c']) => replace('^\(|\)$|\).$|\.$', '') => normalize-space()`. Sometimes MARC field values have extra spaces at the end. 
3. This is just an organization thing, but generally when there are multiple templates for one field, the first template is work, then expression, then manifestation, then additional entities. Not a big deal though, and I don't believe I had mentioned this beforehand. 
4. Another organization thing for test_input and test_output is that the test_output file is generally named [fieldnum]Test-RDA.xml so 255Test-RDA.xml. It also looks like you have quite a few output test files that are more or less the same, so it may be worth trimming those down to one. 
5. Finally, it looks like you found an example where $d and $e have parentheses: 
```xml
            <marc:subfield code="a">Scale 88 mm per 1°</marc:subfield>
            <marc:subfield code="d">(RA 16 hr./Decl. +30° ;</marc:subfield>
            <marc:subfield code="e">eeq. 1973.50).</marc:subfield>
```
It may be worth checking with the mappers on whether these should be removed, since it makes the output look a little funny: 

```xml
      <rdaw:P10082>(RA 16 hr./Decl. +30° ;</rdaw:P10082><!--rdaw:P10082 = has right ascension and declination-->
      <rdaw:P10214>eeq. 1973.50).</rdaw:P10214><!--rdaw:P10214 = has equinox-->
```
My guess is that we will want to remove any initial parentheses or ending parentheses that only come before periods or spaces. We may also want to remove ending semi-colons such as in subfield d. We can ask @pennylenger and @dchen077. 

Overall, really good work! Let's see what Penny and Doreen say about $d and $e and then once `contains()` is changed to `matches()` you should be set to mark this as done!

**Comment by pennylenger** [comment]
Yes, I agree that it would be better to remove the initial parenthesis in $d and the ending parenthesis in $e, as well as the semi-colon at the end of $d and period at the end of $e. The RDA output is clearer without these punctuations.
```
<rdaw:P10082>RA 16 hr./Decl. +30°</rdaw:P10082>
<rdaw:P10214>eeq. 1973.50</rdaw:P10214>
```

**Comment by cspayne** [comment]
Hi @tynancuw, did this ever get completed? Is it coding still in progress?

**Comment by tynancuw** [comment]
It is still coding in progress -- I have a few more tweaks to add I will do those and push this Friday! Sorry for the delay and thanks for the nudge.

**Comment by tynancuw** [comment]
Hi @cspayne, I have pushed a cleaned up version of this. I have a couple of questions about the output, I'll follow up in an email to find a short (15 min or so) time that we could perhaps discussion it.
